File: Javascript/Subsets.js
var subsets = function(nums){
    const result = []
    const subset = []

    function dfs(i){
        if(i>=nums.length){
            result.push([...subset])
            return
        }

        subset.push(nums[i])
        dfs(i+1)

        subset.pop()
        dfs(i+1)
    }

    dfs(0)
    return result
}

File: Python/Subsets.py
def subsets(nums):
    result = []
    subsets = []

    def dfs(i):
        if i >= len(nums):
            result.append(subsets.copy())
            return
        
        subsets.append(nums[i])
        dfs(i+1)

        subsets.pop()
        dfs(i+1)

    dfs(0)
    return result